src/cli/cli_dataset.cpp

```diff
@@ -355,5 +356,205 @@ ThreadError Dataset::ProcessPendingTimestamp(int argc, char *argv[])
     return error;
 }
 
+ThreadError Dataset::ProcessMgmtSetCommand(int argc, char *argv[])
+{
+    ThreadError error = kThreadError_None;
+    otOperationalDataset dataset;
+    uint8_t tlvs[32];
+    uint8_t index;
+    long value;
+    int length = 0;
+    otIp6Address prefix;
+
+    VerifyOrExit(argc > 0, error = kThreadError_Parse);
+
+    memset(&dataset, 0, sizeof(dataset));
+
+    for (index = 1; index < argc; index++)
+    {
+        if (strcmp(argv[index], "activetimestamp") == 0)
```

Comment:
tlvs is to store octets of option binary. And I have check line 445 to guarantee not go beyond end of tlvs.